web_src/js/modules/toast.js

```diff
@@ -0,0 +1,67 @@
+import {htmlEscape} from 'escape-goat';
+import {svg} from '../svg.js';
+
+const levels = {
+  info: {
+    icon: 'octicon-check',
+    background: 'var(--color-green)',
+    duration: 2000,
+  },
+  warning: {
+    icon: 'gitea-exclamation',
+    background: 'var(--color-orange)',
+    duration: -1, // needs to be clicked away
+  },
+  error: {
+    icon: 'gitea-exclamation',
+    background: 'var(--color-red)',
+    duration: -1, // needs to be clicked away
+  },
+};
+
+// See https://github.com/apvarun/toastify-js#api for options
+async function showToast(message, level, {gravity, position, duration, ...other} = {}) {
+  if (!message) return;
+
+  const {default: Toastify} = await import(/* webpackChunkName: 'toastify' */'toastify-js');
+  const {icon, background, duration: levelDuration} = levels[level ?? 'info'];
+
+  const toast = Toastify({
+    text: `
+      <div class='toast-icon'>${svg(icon)}</div>
+      <div class='toast-body'>${htmlEscape(message)}</div>
+      <button class='toast-close'>${svg('octicon-x')}</button>
+    `,
+    escapeMarkup: false,
+    gravity: gravity ?? 'top',
+    position: position ?? 'center',
+    duration: duration ?? levelDuration,
+    style: {background},
+    ...other,
+  });
+
+  toast.showToast();
+
+  toast.toastElement.querySelector('.toast-close').addEventListener('click', () => {
+    toast.removeElement(toast.toastElement);
+  });
+}
+
+export async function showInfo(message, opts) {
```

Comment:
Could the name be `showToastInfo / showToastWarning / showToastError`? It could be clearer when reading `              showError(resp.message);`

Reply:
Yes, I toyed with the idea of naming it `showInfoToast` etc.

Reply:
Oh .... I didn't see the rename in latest commit, it's still called `showInfo` 🤣

Reply:
Done in https://github.com/go-gitea/gitea/pull/25449/commits/34bc1567d0f31fa7f9b200f213aeeb45e6b275e0.